weave-js/src/components/PagePanelComponents/Home/Browse3/typeViews/Audio/AudioPlayer.tsx

```diff
@@ -21,12 +21,25 @@ export const AudioPlayer: FC<{
 }> = ({entity, project, data}) => {
   const {useFileContent} = useWFHooks();
   const [audioUrl, setAudioUrl] = useState<string | null>(null);
+  const [audioFileName, setAudioFileName] = useState<string | null>(null);
+
+  // Find the first audio file in the files object
+  const audioFile = Object.keys(data.files).find(fileName =>
+    fileName.startsWith('audio.')
+  );
+
   const audioBinary = useFileContent({
     entity,
     project,
-    digest: data.files['audio.wav'],
+    digest: audioFile ? data.files[audioFile] : '',
```

Comment:
I think we might want to memo this. If the component gets rendered again it might re-query depending on how data.files is defined

Reply:
Good call, I changed it to Memoize the fileContentParams

I notice for traces with a lot of audio files there's still a bit of pop-in but I think it was pre-existing...